lib/global-styles.php

```diff
@@ -492,6 +492,16 @@ function gutenberg_experimental_global_styles_resolver_styles( $block_selector,
 	$css_rule         = '';
 	$css_declarations = '';
 
+	// Transformations for block-supports strings.
+	foreach ( $block_supports as $block_support ) {
+		// Convert camelCase to kebab-case and add to array.
+		$block_supports[] = strtolower( preg_replace( '/([a-z0-9]|(?=[A-Z]))([A-Z])/', '$1-$2', $block_support ) );
+	}
+	// When adding the kebab-case supports, we didn't check if the item exists in the array
+	// to avoid constly in_array() calls. We'll need to remove the duplicates here.
+	// This is the same as array_unique(), but faster.
+	$block_support = array_flip( array_flip( $block_supports ) );
```

Comment:
What is in the array that is being flipped? This method only works for some types of data. If it is strings, please add a comment to that effect.

Reply:
It is strings, yes. Changed the wording :+1: